test/jdk/javax/xml/jaxp/datatype/8033980/SerializationTest.java

```diff
@@ -25,160 +25,314 @@
-            fail("testReadCal: " + ex.getMessage());
-        }
-    }
 
-    public void testReadDuration() {
-        try {
-            for (String javaVersion : JDK) {
-                Duration d1 = (Duration) fromFile(
-                        javaVersion + FILENAME_DURATION);
-                if (!d1.toString().equalsIgnoreCase(EXPECTED_DURATION)) {
-                    fail("Java version: " + javaVersion
-                            + "\nExpected: " + EXPECTED_DURATION
-                            + "\nActual: " + d1.toString());
-                } else {
-                    success("testReadDuration: read " + javaVersion + " serialized file, passed.");
+                @Override
+                public byte[] getDurationBytes() {
+                    return baos2.toByteArray();
                 }
-            }
-        } catch (ClassNotFoundException ex) {
-            fail("testReadDuration: " + ex.getMessage());
-        } catch (IOException ex) {
-            fail("testReadDuration: " + ex.getMessage());
+
+                @Override
+                public String getDurationBase64() {
+                    return base64dur;
+                }
+            };
+            // To create the pseudocode for <JDK version>GregorianCalendarAndDurationSerData.java for specific version
+            // of JDK (other than 6, 7, 8 and 9), execute this test with that specific JDK version after uncommenting
+            // the below methods calls :
+
+            // generatePseudoCodeForGregCalSerBytes(baos);
+            // generatePseudoCodeForGregCalSerBytesAsBase64(base64);
+            // generatePseudoCodeForDurationSerBytes(baos2);
+            // generatePseudoCodeForDurationSerBytesAsBase64(base64dur);
         }
     }
 
     /**
-     * Create test files
-     *
-     * @param javaVersion JDK version
+     * Provide data for JDK version and Gregorian Calendar serialized bytes.
+     * @return A two-dimensional Array of objects where each element is an array of size three. First element contain JDK version,
+     * second element contain object reference to GregorianCalendarAndDurationSerData specific to JDK version
+     * and third element contain expected Gregorian Calendar as string.
      */
-    public void createTestFile(String javaVersion) {
-        try {
-            DatatypeFactory dtf = DatatypeFactory.newInstance();
-            XMLGregorianCalendar c = dtf.newXMLGregorianCalendar(EXPECTED_CAL);
-            Duration d = dtf.newDuration(EXPECTED_DURATION);
-            toFile((Serializable) c, filePath + javaVersion + FILENAME_CAL);
-            toFile((Serializable) d, filePath + javaVersion + FILENAME_DURATION);
-        } catch (Exception e) {
-            fail(e.getMessage());
-        }
+    @DataProvider(name = "GregorianCalendarData")
+    public Object [][] gregorianCalendarDataBytes() {
+        return new Object [][] {{System.getProperty("java.version"), gregorianCalendarAndDurationSerData[0], EXPECTED_CAL},
```

Comment:
We really only need to call `System.getProperty("java.version")` once in the program and then save off its return value